### ISSUE TITLE
Model parent nodes accurately

### ISSUE THREAD

**Issue by Wilfred** [Wilfred/difftastic#320]
See the [accurate_ancestors_with_fallback](https://github.com/Wilfred/difftastic/compare/accurate_ancestors_with_fallback) branch.

**Comment by Wilfred** [comment]
478e574141ea8680859e52acc61d8b3a823563ef is probably also effective, but probably slower.

**Comment by Wilfred** [comment]
Rebased in accurate_ancestors_v2.

**Comment by Wilfred** [comment]
This produces substantially nicer results, but explores many more nodes. DFT_GRAPH_LIMIT needs to be 10,000,000 for several of the sample files.

**Comment by Wilfred** [comment]
```
 2022-07-20T05:19:32.103Z INFO  difft::options > CLI arguments: ["sample_files/javascript_before.js", "sample_files/javascript_after.js"]
 2022-07-20T05:19:32.136Z INFO  difft::diff::dijkstra > LHS nodes: 150 (1 toplevel), RHS nodes: 329 (1 toplevel)
 2022-07-20T05:19:38.575Z DEBUG difft::diff::dijkstra > Found predecessors for 7323767 vertices (hashmap key: 8 bytes, value: 16 bytes), with 38695 left on heap.
```

This is surprising considering how few nodes there are here.

**Comment by Wilfred** [comment]
If two nodes are equal and unique, does it make sense to consider other possibilities? Could we optimise in that case?

**Comment by Wilfred** [comment]
Perhaps only consider additional parents when the subtree is small?

**Comment by Wilfred** [comment]
This produces a roughly 5x slowdown on my test files, and an increase in peak hashmap size from 1.3M to 7-8M nodes.

**Comment by Wilfred** [comment]
Fixed in c9578185148e04e99497dee9faac902480be0e48.
